### PR TITLE
Add David Stone's thought on variant dissimilarities

### LINKED ISSUE #178: Add discussion of how indirect models or does not model variant (hash, comparisons etc)

### PR DESCRIPTION
**Pull request by jbcoe**
Fixes #178

### CODE REVIEW

File: DRAFT.md [hunk 1]
```diff
@@ -275,6 +275,23 @@ propagates const and is allocator aware.
   (it could be an instance of a derived type). As a result `polymorphic` cannot
   forward comparison operators, hash or formatting to the owned object.
 
+#### Disimilarites with variant and the valueless state
```
Comment (nbx8) [inline DRAFT.md:278]: Dissimilarities

File: DRAFT.md [hunk 2]
```diff
@@ -275,6 +275,23 @@ propagates const and is allocator aware.
...
+
+The sum type, `variant<Ts...>`, models one of several alternatives;
```
Comment (nbx8) [inline DRAFT.md:280]: models one of several alternative types;

File: DRAFT.md [hunk 3]
```diff
@@ -275,6 +275,23 @@ propagates const and is allocator aware.
...
+`indirect<T>` models a single type `T`, but with different storage constraints.
+
+A variant can, like `indirect`, get into a valueless state. For variant, this
+valueless state is accessible when an exception is thrown when changing the
+type: variant has `bool valueless_by_exception()`. When all of the types, `Ts`,
+are comparable, `variant<Ts...>` supports comparison without preconditions: it
+is valid to compare variants when they are in a valueless state. Comparison of
+variant can account for the valueless state with zero-cost. A variant must check
```
Comment (nbx8) [inline DRAFT.md:288]: zero cost

File: DRAFT.md [hunk 4]
```diff
@@ -275,6 +275,23 @@ propagates const and is allocator aware.
   (it could be an instance of a derived type). As a result `polymorphic` cannot
   forward comparison operators, hash or formatting to the owned object.
 
+#### Dissimilarities with variant and the valueless state
+
+The sum type, `variant<Ts...>`, models one of several alternatives;
+`indirect<T>` models a single type `T`, but with different storage constraints.
+
+A variant can, like `indirect`, get into a valueless state. For variant, this
```
Comment (nbx8) [inline DRAFT.md:283]: Like `indirect`, a variant can get into a valueless state.

**Review by nbx8 — DISMISSED** [review]

File: DRAFT.md [hunk 5]
```diff
@@ -275,6 +275,23 @@ propagates const and is allocator aware.
   (it could be an instance of a derived type). As a result `polymorphic` cannot
   forward comparison operators, hash or formatting to the owned object.
 
+#### Similarities and dissimilarities with variant
```
Comment (nbx8) [inline DRAFT.md:278]: Comparison with variant (?)

**Review by nbx8 — APPROVED** [review]

File: DRAFT.md (hunk 5 above)
Comment (jbcoe) [reply DRAFT.md:278]: It's ambiguous as were talking about comparison and indirect == variant could be a thing.

**Review by nbx8 — APPROVED** [review]

**Review by Twon — APPROVED** [review]

**Pull request merged**